### PR TITLE
[JIT] remove prim::SetAttr from list of ops with side effects

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68311

Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

### CODE REVIEW

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

File: test/jit/test_dce.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                y = torch.rand([2, 2])
+                t1 = Thing1()
+                t1.x = x
+                return y
+
+        unscripted = Thing2()
+
+        t2 = torch.jit.script(unscripted)
+        t2.eval()
+        t2 = torch.jit.freeze(t2)
```
Comment (eellison) [inline test/jit/test_dce.py:43]: i guess dce only triggers here after freezing because of inlining ?

**Review by eellison — APPROVED** [review]
nice !!

File: test/jit/test_dce.py (hunk 1 above)
Comment (davidberard98) [reply test/jit/test_dce.py:43]: Yeah, I think it's because `t1.__init__()` is called, and once the init call gets inlined DCE can occur

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

**Pull request closed without merging**